src/Agent.Worker/TestResults/ResultsCommandExtension.cs

```diff
@@ -500,6 +518,41 @@ private void AddTargetBranchInfoToRunCreateModel(RunCreateModel runCreateModel,
                 runCreateModel.BuildReference.TargetBranchName = pullRequestTargetBranchName;
             }
         }
+
+        private async Task PublishEventsAsync(ICustomerIntelligenceServer ciService)
+        {
+            try
+            {
+                CustomerIntelligenceEvent ciEvent = new CustomerIntelligenceEvent()
+                {
+                    Area = _telemetryArea,
+                    Feature = _telemetryFeature,
+                    Properties = _telemetryProperties
```

Comment:
should we make this also as input ? So that we can reuse.

Reply:
I have made telemetryProperties class level variable so that any new properties can be directly added to it, and will be published at last.